src/tpm.rs

```diff
@@ -436,6 +443,157 @@ pub(crate) fn read_mask(mask: &str) -> Result<Vec<PcrSlot>> {
     Ok(pcrs)
 }
 
+// This encodes a quote string as input to Python Keylime's quote checking functionality.
+// The quote, signature, and pcr blob are concatenated with ':' separators. To match the
+// expected format, the quote, signature, and pcr blob must be individually compressed
+// with zlib at the default compression level and then base64 encoded before concatenation.
+//
+// Reference:
+// https://github.com/keylime/keylime/blob/2dd9e5c968f33bf77110092af9268d13db1806c6 \
+// /keylime/tpm/tpm_main.py#L964-L975
+pub(crate) fn encode_quote_string(
+    att: TPM2B_ATTEST,
+    sig: Signature,
+    pcrs_read: PcrSelectionList,
+    pcr_data: PcrData,
+) -> Result<String> {
+    // marshal structs to vec in expected formats. these formats are
+    // dictated by tpm2_tools.
+    let att_vec = &att.attestationData[0..att.size as usize];
+    let sig_vec = sig_to_vec(sig.try_into()?);
+    let pcr_vec = pcrdata_to_vec(pcrs_read, pcr_data);
+
+    // zlib compression
+    let mut att_comp = ZlibEncoder::new(Vec::new(), Compression::default());
```

Comment:
nit: This may be something to turn into a function since the same steps happen multiple times. Not a required change though.

Reply:
Could do in a future commit!